frontend/src/metabase/parameters/utils/cards.ts

```diff
@@ -10,57 +10,70 @@ import {
   getValuePopulatedParameters,
   hasParameterValue,
 } from "metabase/parameters/utils/parameter-values";
+import { ParameterWithTarget } from "metabase/parameters/types";
+import { Parameter, ParameterTarget } from "metabase-types/types/Parameter";
+import { Card } from "metabase-types/types/Card";
+import { TemplateTag } from "metabase-types/types/Query";
+import Metadata from "metabase-lib/lib/metadata/Metadata";
 
-// NOTE: this should mirror `template-tag-parameters` in src/metabase/api/embed.clj
-export function getTemplateTagParameters(tags) {
-  function getTemplateTagType(tag) {
-    const { type } = tag;
-    if (type === "date") {
-      return "date/single";
-    } else if (type === "string") {
-      return "string/=";
-    } else if (type === "number") {
-      return "number/=";
-    } else {
-      return "category";
-    }
+function getTemplateTagType(tag: TemplateTag) {
+  const { type } = tag;
+  if (type === "date") {
+    return "date/single";
+    // @ts-expect-error -- preserving preexisting incorrect types (for now)
+  } else if (type === "string") {
```

Comment:
`type` can't be a `"string"`. I think this ought to be `"text"`, but we can address later. `"text"` template tag parameters fall through to the `else` and end up with a `"category"` type, which still works.